### PR TITLE
Improve usability for conditional seasonality

### LINKED ISSUE #609: New feature: Utility to add discrete regressors for quarter/month of year, weekday, day of week, quarter/hour of day
Utility to make it simple to add discrete regressors for different typical 'seasons' of a time-series.
See #607 for example.

### PR DESCRIPTION
**Pull request by judussoari**
## :crystal_ball: Key changes

Extends #1067 
Added docstring example, a new tutorial notebook for conditional seasonality, as well as some validation for float values between 0..1 in the conditional seasonality column.
Also, added helper function to integrate four seasons and/or weekday/weekend conditions. Fixes #609 

## :clipboard: Review Checklist
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, added docstrings and data types to function definitions.
- [x] I have added pytests to check whether my feature / fix works.

### CODE REVIEW

**Comment by judussoari** [comment]
@ourownstory when you or someone else review this, please have a look at the new helper function with the pre-implmented conditional seasonalities, and evaluate whether it makes sense

**Comment by judussoari** [comment]
@ourownstory when you or someone else review this, please have a look at the new helper function with the pre-implmented conditional seasonalities, and evaluate whether it makes sense

File: neuralprophet/df_utils.py [hunk 1]
```diff
@@ -1531,3 +1531,72 @@ def join_dfs_after_data_drop(predicted, df, merge=False):
...
+        # spring --> summer:
+        df.loc[(df["ds"].dt.month == 5) & (df["ds"].dt.day == 31), "spring"] = soft_transition
+        df.loc[(df["ds"].dt.month == 6) & (df["ds"].dt.day == 1), "spring"] = soft_transition
+        df.loc[(df["ds"].dt.month == 5) & (df["ds"].dt.day == 31), "summer"] = 1 - soft_transition
+        df.loc[(df["ds"].dt.month == 6) & (df["ds"].dt.day == 1), "summer"] = 1 - soft_transition
+
+    return df
+
+
+def add_weekend_condition(df):
```
Comment (ourownstory) [inline neuralprophet/df_utils.py:1586]: maybe rename to weekday - make weekdays 1

File: neuralprophet/df_utils.py [hunk 2]
```diff
@@ -1531,3 +1531,72 @@ def join_dfs_after_data_drop(predicted, df, merge=False):
         return predicted, df
     else:
         return df_merged.rename_axis("ds").reset_index()
+
+
+def add_four_seasons_condition(df, soft_transition=0):
```
Comment (ourownstory) [inline neuralprophet/df_utils.py:1536]: transition_period: steps to linearly interpolate between the two

Comment (ourownstory) [reply neuralprophet/df_utils.py:1536]: note: Northern hemisphere

File: neuralprophet/forecaster.py [hunk 3]
```diff
@@ -648,6 +648,28 @@ def add_seasonality(self, name, period, fourier_order, condition_name=None):
                 number of Fourier components to use.
             condition_name : string
                 string name of the seasonality condition.
+
+        Examples
+        --------
+        Adding a quarterly changing weekly seasonality to the model. First, add columns to df.
+        The columns should contain only zeros and ones (or floats), deciding when to apply seasonality.
+            >>> df["summer_week"] = df["ds"].apply(lambda x: x.month in [6, 7, 8])
```
Comment (ourownstory) [inline neuralprophet/forecaster.py:656]: "summer_week", can drop week

File: neuralprophet/forecaster.py [hunk 4]
```diff
@@ -3176,3 +3211,34 @@ def conformal_predict(
             c.plot(plotting_backend)
 
         return df
+
+    def add_condition_to_df(self, df, condition="four_seasons", soft_transition=0):
```
Comment (ourownstory) [inline neuralprophet/forecaster.py:3215]: I think we can use the df_utils directly!

File: neuralprophet/df_utils.py (hunk 2 above)
Comment (ourownstory) [reply neuralprophet/df_utils.py:1536]: quarter instead of four seasons?

**Review by ourownstory — CHANGES_REQUESTED** [review]
Great work!

RE:tutorial:

- mention helper functions
- explain what a condition is/must be
- plot_params: just plot Seasonalities

**Review by karl-richter — APPROVED** [review]
Looks good!

**Pull request merged**